juju/juju
# [JUJU-2240] disallow migration or upgrade of a model to 3.1 if charm store charms used in model

3.1 will no support charm store charms. Ensure that none exist in models migrating or upgrading to 3.1.

## QA steps

Bootstrap a juju 3.0 controller and add a model with charmhub, local and charm store charms to it. Try to upgrade to 3.1, and see if fail. Remove the charmstore application from the model and try again. This time succeeding.

Bootstrap a juju 3.0  and 3.1 controllers. Add a model to the 3.0 controller with charmhub, local and charm store charms to it. Try to migrate it to 3.1, and see if fail. Remove the charmstore application from the model and try again. This time succeeding.

Try other scenarios of upgrade/migration as well.
